### DISCUSSION TITLE
Mechanic talent "Residual Waste" and "Mass Production" should check a generic tag to prevent item duplication

### DISCUSSION THREAD

**Discussion by A2ur3** [FakeFishGames/Barotrauma#15246] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

EK crate items
for example` Crate of Plastic`
fabrication recipe = 10 plastic -> can be triggered by  `Mass Production`
deconstruction outputs = 10 plastic -> can be triggered by `Residual Waste` unless you add one of these tags to this item `"geneticmaterial,unidentifiedgeneticmaterial,circuitboxcomponent,lightcomponent"`

### Reproduction steps

1. subscribe and enable ` EK Utilities Remastered`
2. make a character who has talents in question fabricate and deconstruct `Crate of Plastic`
3. item duplication

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

v1.6.19.1 (Unto the Breach Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Ek-Videogames** [comment]
Honestly, if the devs ever felt like taking the initiative to try and fix & re-implement recycling into the game again. They could have some kind of skill and/or talent based mechanic for increasing the amount of items yielded from recycling, as opposed to the sort of hacky half-measured implementation of spawning generic scrap.

For example: Having deconstruction results defined in-XML with some kind of 'minimum' and 'maximum' values. 
(or like how 'deconstructionCondition' already works) 

Example

>     <Deconstruct time="20">
>       <Item identifier="aluminium" amount="1" maxamount="4"/>
>       <Item identifier="copper" amount="1" maxamount="2"/>
>       <Item identifier="tin" amount="0" maxamount="2"/>
>       <Item identifier="plastic" amount="0" maxamount="1"/>
>     </Deconstruct>

Could be done several ways, like requiring talent, or some skill level.

**Comment by Regalis11** [comment]
Addressed in https://github.com/FakeFishGames/Barotrauma-development/commit/a09af1e4b432780a3d32ea9bd158fd1e85325ee6

> Made "residual waste" and "mass production" talents a bit more modder-friendly. The former had a chance of duplicating deconstruction output and the latter a chance of not consuming some of the fabrication ingredients. It was hard to exclude items from those talents though: addressed that by making them ignore items with the tags "disallowduplicatedeconstructoutput" and "disallowremovefabricationingredient" respectively. #15246

The change unfortunately didn't make it in time for the upcoming update, but will be included in the one after that.

**Comment by A2ur3** [comment]
@Regalis11 I'm afraid the change won't solve my problem
we have a item `Crate of Plastic` that can be fabricated from 10 plastic and be deconstructed into 10 plastic
I can add `disallowduplicatedeconstructoutput` tag to this item to prevent item duplication by `residual waste`
but to prevent item duplication from `mass production` I need to add `disallowremovefabricationingredient` to `Plastic`
This prevents other recipes triggering the talent

**Reply by Regalis11** [reply]
Hmm, that's true. Do you have any ideas on how else the problem could be addressed?

**Reply by A2ur3** [reply]
🤷‍♂️

**Comment by Ydrec** [comment]
Someone got this problem again recently
This could be fixed by adding new property to fabricationRecipe like disableTalentsOnInput and disableTalentsOnOutput
  
  and then add a check for these properties before
https://github.com/FakeFishGames/Barotrauma/blob/783499f152669a89745e4f9fa46167b6f2bf5195/Barotrauma/BarotraumaShared/SharedSource/Items/Components/Machines/Fabricator.cs#L448
  and
https://github.com/FakeFishGames/Barotrauma/blob/783499f152669a89745e4f9fa46167b6f2bf5195/Barotrauma/BarotraumaShared/SharedSource/Items/Components/Machines/Fabricator.cs#L471

**Reply by A2ur3** [reply]
I suggest that player-crafted items should not trigger the “Residual Waste” talent, in order to prevent players from repeatedly crafting and deconstructing items to farm materials. This is already feasible in the vanilla game, as seen with items like the Pulse Tri-Laser Fuel Box
